### PR TITLE
[OpenMPI] enable Fortran

### PR DESCRIPTION
**Pull request by vchuravy**
Needed for WI4MPI

### CODE REVIEW

**Comment by giordano** [comment]
CC: @lcw, I think he looked into this before?

**Comment by lcw** [comment]
Yes, cross compiling OpenMPI is a pain.  The autoconf tooling has many runtime checks of the fortran abi.  It would be great if we could get it to work.  Another option is to use spack to build WI4MPI as I did [here](https://gist.github.com/lcw/c89df81da73adb003f0f89aa123ee225).

File: O/OpenMPI/build_tarballs.jl [hunk 1]
```diff
@@ -21,8 +22,8 @@ fi
     --host=${target} \
     --enable-shared=yes \
     --enable-static=no \
-    --disable-mpi-fortran \
-    --without-cs-fs
+    --without-cs-fs \
+    # --with-cross=${WORKSPACE}/srcdir/${target}
```
Comment (giordano) [inline O/OpenMPI/build_tarballs.jl:26]:
```suggestion
    --with-cross=${WORKSPACE}/srcdir/${target}
```
?

**Comment by giordano** [comment]
The license file isn't installed automatically anymore, need to do it manually

**Comment by giordano** [comment]
So files are all equal apart from aarch64-apple-darwin which has
```patch
+
+ompi_cv_fortran_alignment_type_test_mpi_handle_='    4'
+ompi_cv_fortran_sizeof_type_test_mpi_handle_='4'
```
Is this relevant for the other platforms?  Any idea why only this platforms failed without them? :sweat_smile:

**Comment by vchuravy** [comment]
> Is this relevant for the other platforms? Any idea why only this platform failed without them?

It was in the original file https://github.com/open-mpi/ompi/files/6737172/aarch64-linux-gcc.txt but I removed it after it wasn't needed anywhere else.
But otherwise no clue...

**Pull request merged**